mesh/meshdb.go

```diff
@@ -41,15 +41,8 @@ type DB struct {
 }
 
 // NewPersistentMeshDB creates an instance of a mesh database.
-func NewPersistentMeshDB(path string, blockCacheSize int, logger log.Log) (*DB, error) {
-	if err := os.MkdirAll(path, os.ModePerm); err != nil {
-		return nil, fmt.Errorf("failed to create %s: %w", path, err)
-	}
-	db, err := sql.Open("file:" + filepath.Join(path, "state.sql"))
-	if err != nil {
-		return nil, fmt.Errorf("open sqlite db %w", err)
-	}
-
+func NewPersistentMeshDB(db *sql.Database, blockCacheSize int, logger log.Log) (*DB, error) {
+	var err error
```

Comment:
can we remove it and use err below in if statement? seemed like it hangs in some weird way

Reply:
Sure, good idea